### PR TITLE
Omit "Accessing site insecurely via HTTP." warning if HTTP is secure enough

### PR DESCRIPTION
**Pull request by Al2Klimov**
## Summary

This is the case for localhost and Tor hidden services.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by Al2Klimov** [comment]
Cool that you consider this for NC 27. https://github.com/nextcloud/server/pull/37330#event-8824318862

But IMAO there’s some a little more important of mine, a literal security hole IMAO:

* #29423

**Review by artonge — APPROVED** [review]
Makes sense

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -717,7 +717,7 @@
 							type: OC.SetupChecks.MESSAGE_TYPE_WARNING
 						});
 					}
-				} else {
+				} else if (!/(?:^(?:localhost|127\.0\.0\.1|::1)|\.onion)$/.exec(window.location.hostname)) {
```
Comment (Pytal) [inline core/js/setupchecks.js:720]: Could we add tests for the cases where the "Accessing site insecurely via HTTP" message is returned?

Comment (Pytal) [reply core/js/setupchecks.js:720]: Makes sense, then please adapt the removed test for when non-localhost IPs are accessed insecurely :)

Comment (Al2Klimov) [reply core/js/setupchecks.js:720]: Are they accessed at all? If they were, I wouldn’t have to remove the test I guess.

Comment (Al2Klimov) [reply core/js/setupchecks.js:720]: If you **really want** that test, I _could_ drop the localhost handling and just keep the Tor hidden services. Almost no one actually uses NC in prod via localhost.

Comment (Pytal) [reply core/js/setupchecks.js:720]: Only nice-to-have I guess so current state is fine with me @Al2Klimov :)

**Review by Pytal — APPROVED** [review]

**Pull request merged**